src/Symfony/Component/Workflow/Dumper/GraphvizDumper.php

```diff
@@ -0,0 +1,204 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Workflow\Dumper;
+
+use Symfony\Component\Workflow\Definition;
+use Symfony\Component\Workflow\Marking;
+use Symfony\Component\Workflow\Transition;
+
+/**
+ * GraphvizDumper dumps a workflow as a graphviz file.
+ *
+ * You can convert the generated dot file with the dot utility (http://www.graphviz.org/):
+ *
+ *   dot -Tpng workflow.dot > workflow.png
+ *
+ * @author Fabien Potencier <fabien@symfony.com>
+ * @author Grégoire Pineau <lyrixx@lyrixx.info>
+ */
+class GraphvizDumper implements DumperInterface
+{
+    private static $defaultOptions = array(
+        'graph' => array('ratio' => 'compress', 'rankdir' => 'LR'),
+        'node' => array('fontsize' => 9, 'fontname' => 'Arial', 'color' => '#333333', 'fillcolor' => 'lightblue', 'fixedsize' => true, 'width' => 1),
+        'edge' => array('fontsize' => 9, 'fontname' => 'Arial', 'color' => '#333333', 'arrowhead' => 'normal', 'arrowsize' => 0.5),
+    );
+
+    /**
+     * Dumps the workflow as a graphviz graph.
+     *
+     * Available options:
+     *
+     *  * graph: The default options for the whole graph
+     *  * node: The default options for nodes (places + transitions)
+     *  * edge: The default options for edges
+     *
+     * {@inheritdoc}
+     */
+    public function dump(Definition $definition, Marking $marking = null, array $options = array())
+    {
+        $places = $this->findPlaces($definition, $marking);
+        $transitions = $this->findTransitions($definition);
+        $edges = $this->findEdges($definition);
+
+        $options = array_replace_recursive(self::$defaultOptions, $options);
+
+        return $this->startDot($options)
+            .$this->addPlaces($places)
+            .$this->addTransitions($transitions)
+            .$this->addEdges($edges)
+            .$this->endDot();
+    }
+
+    private function findPlaces(Definition $definition, Marking $marking = null)
+    {
+        $places = array();
+
```

Comment:
With what ? You added 4 times the same comments, so I thinks this "inconsistency" is consistent. And Anyway, I prefer like that ;)